src/librustc_mir/interpret/eval_context.rs

```diff
@@ -860,30 +854,18 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     }
 
     pub fn generate_stacktrace(&self, explicit_span: Option<Span>) -> Vec<FrameInfo<'tcx>> {
-        let mut last_span = None;
         let mut frames = Vec::new();
         for frame in self.stack().iter().rev() {
-            // make sure we don't emit frames that are duplicates of the previous
-            if explicit_span == Some(frame.span) {
-                last_span = Some(frame.span);
-                continue;
-            }
-            if let Some(last) = last_span {
-                if last == frame.span {
-                    continue;
-                }
-            } else {
-                last_span = Some(frame.span);
-            }
```

Comment:
I removed this because I don't think backtraces usually deduplicate frames. It just seems confusing.

Reply:
I added the deduplication because when errors that happen due to infinite recursion were quite verbose. I guess that didn't work for mutual recursion anyway... so let's go with full backtraces.